src/client/common/variables/environmentVariablesProvider.ts

```diff
@@ -104,6 +87,23 @@ export class EnvironmentVariablesProvider implements IEnvironmentVariablesProvid
             this.disposables.push(envFileWatcher.onDidDelete(() => this.onEnvironmentFileChanged(workspaceFolderUri)));
         }
     }
+    private async _getEnvironmentVariables(resource?: Uri): Promise<EnvironmentVariables> {
+        let customEnvVars = await this.getCustomEnvironmentVariables(resource);
+        if (customEnvVars) {
```

Comment:
This looks wrong. I think that you need a ! here?

Reply:
Is that why you also have customEnvVars! below here in the if check? Those shouldn't be needed if this if a ! here.

Reply:
Yeah this looks backwards to me too.

Reply:
Yup, explains why all tests are failing.